internal/jobobject/jobobject.go

```diff
@@ -503,6 +507,27 @@ func (job *JobObject) ApplyFileBinding(root, target string, merged bool) error {
 	return nil
 }
 
+// isOpenedJobSilo is a helper to determine if a job object that was opened is a silo. This should ONLY be called
+// from `Open` and any callers in this package afterwards should use `job.isSilo()``
```

Comment:
Could also consider using a `sync.Once` and doing the query inside `isSilo`. Then you just have a single function that is called to determine if the job is a silo.

Reply:
Let me play around with this

Reply:
Main reason I didn't want to have one method is the query is mostly useless besides for opening as if we're the ones making the silo we already know it is if the upgrade succeeded so we can just check a bool instead of querying

Reply:
I'm semi-following, however how would a user use isSilo un-exported?

Reply:
Also for the first part we'd intentionally left the handle not accessible, it's not exposed in anyway to a user of the package today. That'd be a pretty drastic change in API style

Reply:
ohh, opps
I guess they would have to be within this package ...

Reply:
I'm going to leave this as is for now. Kevin and I had talked and arrived that things would be easier for this scenario if the only way that you could upgrade to a silo was at creation time, as there's no need to maintain any sort of thread safe way to tell if we're a silo or not. I'd love to move the jobobject package to winio, so maybe if that comes to fruition we can take a longer look at the API here and adjust accordingly. For our needs here at least, I don't think we actually need to maintain a way to upgrade to a silo *after* creation.